Change - 3239: Move docker volumes settings

Poltergeist opened the pull request:
This moves the docker container volumes setting from the container section to the volumes section. Docker volumes are only shown if a docker image is set.

![image](https://cloud.githubusercontent.com/assets/156010/13113540/b96762f8-d58f-11e5-857e-d1ad43870adf.png)

![image](https://cloud.githubusercontent.com/assets/156010/13113556/c5a22a1c-d58f-11e5-9623-1c759a93d697.png)


Closes mesosphere/marathon#3239

aldipower commented on src/js/components/OptionalContainerVolumesComponent.jsx:
```diff
@@ -0,0 +1,146 @@
+    if (dockerImage == null || dockerImage === "") {
+      return (
+        <div className="warning">
+          Add docker image to add docker volumes.
```
"Add a Docker Image to add Docker Volumes."

aldipower commented on src/js/components/OptionalContainerVolumesComponent.jsx:
```diff
@@ -0,0 +1,146 @@
+    return (
+      <div>
+        <h4>
+          Docker container Volumes
```
"Docker Container Volumes" or "Docker container volumes"

aldipower commented on src/js/components/OptionalContainerVolumesComponent.jsx:
```diff
@@ -0,0 +1,146 @@
+    var rows = this.state.rows.containerVolumes;
+    var dockerImage = this.props.fields.dockerImage;
+
+    if (dockerImage == null || dockerImage === "") {
```
This works smoothly. :)

aldipower commented on src/js/components/OptionalContainerVolumesComponent.jsx:
```diff
@@ -0,0 +1,146 @@
+
+import ContainerConstants from "../constants/ContainerConstants";
+
+var OptionalVolumesComponent = React.createClass({
```
Oops

Poltergeist replied:
oOps

aldipower commented on src/js/components/OptionalContainerVolumesComponent.jsx:
```diff
@@ -0,0 +1,146 @@
+import ContainerConstants from "../constants/ContainerConstants";
+
+var OptionalVolumesComponent = React.createClass({
+  displayName: "OptionalContainerVolumesComponent",
```
I find the naming confusing, there is also a OptionalVolumesComponent.
What is what? Does it makes sense to combine these?

aldipower commented:
If I add a docker image, then add a docker volume and then remove the image again, everything during editing the form, the docker volumes persists in the background, which leads to a server error response because an invalid docker structure is submitted.

If I remove the docker image, the according docker volumes should be removed too.

Poltergeist commented:
Thanks @aldipower for reviewing this so fast. I tried to combine in one component first. But then decided against it since the duplicable rows mixin is not working smoothly with multiple fields which might have the same name. In this case it was the `containerPath` I could rename the fields to something like `containerVolumesLocalContainerPath` But then we have to take care of the fields in the transforms. I wanted to keep the fields as close as possible to the model. 

For the api error this is one case I meant to test before turning this into an official PR.

aldipower commented:
Ah, sorry, thought this PR is already official. :)

This makes sense to me now @Poltergeist having this into 2 components. I see the point. I am fine with both solutions.
But still I find the name kind of confusing how about `OptinalDockerVolumesCcomponent`? only a suggestion.

aldipower commented:
Good work btw, looks straight away!

Poltergeist commented:
Okay changed the names a bit and split up the optionalVolumesComponent, which now is only a wrapper for different Volume types.
```
OptionalVolumesComponent
 ├─ LocalVolumesComponent
 └─ ContainerVolumesComponent
```
Also I renamed `containerVolumesLocal` to be more consistent to `localVolumes` so now there are only
  - `containerVolumes`
  - `localVolumes`
I think that makes more sense.

aldipower commented on src/js/components/ContainerSettingsComponent.jsx:
```diff
@@ -352,8 +273,7 @@ var ContainerSettingsComponent = React.createClass({
         <div className="duplicable-list">{this.getParametersRows()}</div>
-        <h4>Volumes</h4>
-        <div className="duplicable-list">{this.getVolumesRows()}</div>
+        <div>You can set you Docker volume settings below.</div>
```
`You can set you_r_`

aldipower commented on src/js/components/ContainerVolumesComponent.jsx:
```diff
@@ -0,0 +1,143 @@
+    return (
+      <div key={row.consecutiveKey} className={rowClassSet}>
+        <fieldset className="row duplicable-row"
+          onChange={this.handleChangeContainerVolumes.bind(null, i)}>
```
+2 indent

aldipower commented on src/js/stores/transforms/AppFormModelPostProcess.js:
```diff
@@ -32,6 +32,15 @@ const AppFormModelPostProcess = {
     }
 
+    if ((container.docker == null || container.docker.image == null ||
+      container.docker.image === "") &&
```
+2 indent

aldipower commented:
I like the naming of the components and fields very much now.

aldipower commented:
@Poltergeist Besides some minor stylistic things this looks perfect!

aldipower commented:
:100: Danke!
